File: src/main/java/com/github/sorusclient/client/adapter/v1_8_9/PlayerEntityImpl.java
package com.github.sorusclient.client.adapter.v1_8_9;

import com.github.sorusclient.client.adapter.IPlayerInventory;
import com.github.sorusclient.client.adapter.IPlayerEntity;
import v1_8_9.net.minecraft.entity.Entity;
import v1_8_9.net.minecraft.entity.player.PlayerEntity;

public class PlayerEntityImpl extends LivingEntityImpl implements IPlayerEntity {

    public PlayerEntityImpl(Entity entity) {
        super(entity);
    }

    @Override
    public double getHunger() {
        return ((PlayerEntity) this.entity).getHungerManager().getFoodLevel();
    }

    @Override
    public double getArmorProtection() {
        return ((PlayerEntity) this.entity).getArmorProtectionValue();
    }

    @Override
    public int getExperienceLevel() {
        return ((PlayerEntity) this.entity).experienceLevel;
    }

    @Override
    public double getExperiencePercentUntilNextLevel() {
        return ((PlayerEntity) this.entity).experienceProgress;
    }

    @Override
    public IPlayerInventory getInventory() {
        return new PlayerInventoryImpl(((PlayerEntity) this.entity).inventory);
    }

}

File: src/main/kotlin/com/github/sorusclient/client/adapter/v1_8_9/PlayerEntityImpl.kt
package com.github.sorusclient.client.adapter.v1_8_9

import com.github.sorusclient.client.adapter.IPlayerEntity
import com.github.sorusclient.client.adapter.IPlayerInventory
import v1_8_9.net.minecraft.entity.Entity
import v1_8_9.net.minecraft.entity.player.PlayerEntity

class PlayerEntityImpl(entity: Entity) : LivingEntityImpl(entity), IPlayerEntity {
    override val hunger: Double
        get() = (entity as PlayerEntity).hungerManager.foodLevel.toDouble()
    override val armorProtection: Double
        get() = (entity as PlayerEntity).armorProtectionValue.toDouble()
    override val experienceLevel: Int
        get() = (entity as PlayerEntity).experienceLevel
    override val experiencePercentUntilNextLevel: Double
        get() = (entity as PlayerEntity).experienceProgress.toDouble()
    override val inventory: IPlayerInventory
        get() = PlayerInventoryImpl((entity as PlayerEntity).inventory)
}